codec-http/src/main/java/io/netty/handler/codec/http/cookie/ClientCookieDecoder.java

```diff
@@ -241,13 +244,20 @@ private void parse7(int nameStart, int valueStart, int valueEnd) {
             }
         }
 
-        private void parse8(int nameStart) {
+        private void parse8(int nameStart, int valueStart, int valueEnd) {
             if (header.regionMatches(true, nameStart, CookieHeaderNames.HTTPONLY, 0, 8)) {
                 httpOnly = true;
+            } else if (header.regionMatches(true, nameStart, CookieHeaderNames.SAMESITE, 0, 8)) {
+                String sameSiteValue = computeValue(valueStart, valueEnd);
+                try {
+                    sameSite = SameSite.of(sameSiteValue);
+                } catch (IllegalArgumentException e) {
```

Comment:
IMHO, you don't need to throw an IllegalArgumentException from SameSite.of and catch it right after. You could just return null.